x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/sparse_vector_search.yml

```diff
@@ -0,0 +1,306 @@
+  - do:
+      search:
+        index: index-with-sparse-vector
+        body:
+          query:
+            sparse_vector:
+              field: ml.tokens
+              inference_id: text_expansion_model
+              text: "octopus comforter smells"
+              prune: true
+              pruning_config: { }
+  - match: { hits.total.value: 4 }
+  - match: { hits.hits.0._source.source_text: "the octopus comforter smells" }
+
+---
+"Test sparse_vector search with a pruning configuration that only keeps pruned tokens":
+  - do:
+      search:
+        index: index-with-sparse-vector
+        body:
+          query:
+            sparse_vector:
+              field: ml.tokens
+              inference_id: text_expansion_model
+              text: "octopus comforter smells"
+              prune: true
+              pruning_config:
+                tokens_freq_ratio_threshold: 4
+                tokens_weight_threshold: 0.4
+                only_score_pruned_tokens: true
+  - match: { hits.total.value: 0 }
+
+---
+"Test sparse_vector search with query vector":
+  - do:
+      search:
+        index: index-with-sparse-vector
+        body:
+          query:
+            sparse_vector:
+              field: ml.tokens
+              query_vector:
+                the: 1.0
+                comforter: 1.0
+                smells: 1.0
+                bad: 1.0
+  - match: { hits.total.value: 5 }
+  - match: { hits.hits.0._source.source_text: "the octopus comforter smells" }
+
+---
+"Test sparse_vector search with query vector and pruning config":
+  - do:
+      search:
+        index: index-with-sparse-vector
+        body:
+          query:
+            sparse_vector:
+              field: ml.tokens
+              query_vector:
+                the: 1.0
+                comforter: 1.0
+                smells: 1.0
+                bad: 1.0
+              prune: true
+              pruning_config:
+                tokens_freq_ratio_threshold: 1
+                tokens_weight_threshold: 0.4
+                only_score_pruned_tokens: false
+
+  - match: { hits.total.value: 5 }
+  - match: { hits.hits.0._source.source_text: "the octopus comforter smells" }
+
+---
+"Test sparse_vector search with query vector and pruning config with only score pruned tokens":
+  - do:
+      search:
+        index: index-with-sparse-vector
+        body:
+          query:
+            sparse_vector:
+              field: ml.tokens
+              query_vector:
+                the: 1.0
+                comforter: 1.0
+                smells: 1.0
+                bad: 1.0
+              pruning_config:
+                tokens_freq_ratio_threshold: 1
+                tokens_weight_threshold: 0.4
+                only_score_pruned_tokens: true
+  - match: { hits.total.value: 0 }
+
+---
+"Test sparse_vector only allows one of inference_id or query_vector":
+  - do:
+      catch: /Failed to build \[sparse_vector\] after last required field arrived/
```

Comment:
This isn't a very helpful error message (or at least this part of it isn't). Can we check that we generate a helpful error message here?